services/plugin/src/Plugin.tsx

```diff
@@ -47,13 +47,33 @@ export const Plugin = ({
 
     const [communicationReceived, setCommunicationReceived] =
         useState<boolean>(false)
+    const [prevCommunicationReceived, setPrevCommunicationReceived] =
+        useState<boolean>(false)
 
     const [inErrorState, setInErrorState] = useState<boolean>(false)
 
+    // since we do not know what props are passed, the dependency array has to be keys of whatever is standard prop
+    const memoizedPropsToPass = useMemo(
```

Comment:
this is the fix. I'm not sure I like this solution but since it's unknown what props are actually going to be passed, this was the best way I could come up with handling the dependency array.

Reply:
🤦‍♂️ thanks @kabaros. Yes; this should have been `[...Object.values(propsToPass)]`...but now I think I remember that I did that originally, but then thought that the order of the props wouldn't theoretically guaranteed to be in the same order on the object, hence it might be better to do something like `[...Object.keys(propsToPass).sort().map(k=>propsToPass[k])]`, but then I get a type error:

```Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{ propsToPass: PropsObject; }'.
  No index signature with a parameter of type 'string' was found on type '{ propsToPass: PropsObject; }'.
  ```
  
  And then I tried to explicitly declare propsToPass as an object with string keys (I made an interface of 
  ```
  interface PropsObject {
    [propName: string]: any;
}
```
but that did not resolve the type issues 🫤

I've pushed a change to `[...Object.keys(propsToPass).sort().map(k=>propsToPass[k])]`, but let me know if you have tips about what I could to fix the type warnings 🙏